### PR TITLE
[GenericFilter] Support custom conditions and configurations 2.0.0

### PR DESCRIPTION
**Pull request by KremnevDmitry**

### CODE REVIEW

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/genericfilter/FlowuiDataGenericFilterSupport.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static io.jmix.flowui.component.genericfilter.FilterUtils.generateFilterPath;
+
+@Internal
+public class FlowuiDataGenericFilterSupport extends GenericFilterSupport {
```
Comment (glebfox) [inline jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/genericfilter/FlowuiDataGenericFilterSupport.java:66]: Something odd with Configuration editor. Checkboxes do not handles their label themself, as a result they are misaligned.

Screenshot 2023-05-29 at 17 23 01

See a default defauilt view as a reference:

Screenshot 2023-05-29 at 17 23 09

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/entity/FilterConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import jakarta.persistence.*;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+@JmixEntity
+@Entity(name = "flowui_FilterConfiguration")
+@Table(name = "FLOWUI_FILTER_CONFIGURATION")
+@SystemLevel
+public class FilterConfiguration implements Serializable {
```
Comment (glebfox) [inline jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/entity/FilterConfiguration.java:35]: Support in FilterUrlQueryParameters

**Review by glebfox — CHANGES_REQUESTED** [review]

**Comment by glebfox** [comment]
Some visual bug appears that makes filter filicker:

https://github.com/jmix-framework/jmix/assets/6077149/1c77e14e-ac9d-48f3-8aa0-25767c41dfcf

File: jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/entity/FilterConfiguration.java (hunk 2 above)
Comment (KremnevDmitry) [reply jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/entity/FilterConfiguration.java:35]: To be discussed

**Review by glebfox — APPROVED** [review]

**Pull request merged**